qualtran/cirq_interop/_cirq_interop.py

```diff
@@ -81,28 +100,65 @@ def add_my_tensors(
         incoming: Dict[str, 'SoquetT'],
         outgoing: Dict[str, 'SoquetT'],
     ):
-        unitary = cirq.unitary(self.gate).reshape((2,) * 2 * self.n_qubits)
+        unitary = cirq.unitary(self.gate).reshape((2,) * 2 * self.cirq_registers.total_bits())
+        incoming_list = [
+            *itertools.chain.from_iterable(
+                [np.array(incoming[reg.name]).flatten() for reg in self.signature.lefts()]
+            )
+        ]
+        outgoing_list = [
+            *itertools.chain.from_iterable(
+                [np.array(outgoing[reg.name]).flatten() for reg in self.signature.rights()]
+            )
+        ]
 
         tn.add(
             qtn.Tensor(
-                data=unitary,
-                inds=outgoing['qubits'].tolist() + incoming['qubits'].tolist(),
-                tags=[self.short_name(), tag],
+                data=unitary, inds=outgoing_list + incoming_list, tags=[self.short_name(), tag]
             )
         )
 
     def as_cirq_op(
-        self, qubit_manager: 'cirq.QubitManager', qubits: 'CirqQuregT'
+        self, qubit_manager: 'cirq.QubitManager', **cirq_quregs: 'CirqQuregT'
     ) -> Tuple['cirq.Operation', Dict[str, 'CirqQuregT']]:
-        assert qubits.shape == (self.n_qubits, 1)
-        return self.gate.on(*qubits[:, 0]), {'qubits': qubits}
+        merged_qubits = np.concatenate(
+            [cirq_quregs[reg.name].flatten() for reg in self.signature.lefts()]
+        )
+        assert len(merged_qubits) == cirq.num_qubits(self.gate)
+        return self.gate.on(*merged_qubits), cirq_quregs
 
     def t_complexity(self) -> 'cirq_ft.TComplexity':
         return cirq_ft.t_complexity(self.gate)
 
+    def wire_symbol(self, soq: 'Soquet') -> 'WireSymbol':
+        from qualtran.drawing import directional_text_box
+
+        wire_symbols = cirq.circuit_diagram_info(self.gate).wire_symbols
+        begin = 0
+        symbol = soq.pretty()
+        for reg in self.signature:
+            finish = begin + np.product(reg.shape)
+            if reg == soq.reg:
+                symbol = np.array(wire_symbols[begin:finish]).reshape(reg.shape)[soq.idx]
```

Comment:
wait .. what is happening here? since bitsize is always `1`, we're guaranteed that symbol is one string, yes? maybe add a little `assert` just to elucidate the code

Reply:
Yes, `symbol` (now typed as `str`) is always expected to be a single string. The `reshaping` is to make it convenient to convert from flat list of symbols, one for each qubit, to a multi-dimensional representation so we can use the composite index of `soq.idx` directly. 

`reg.bitsize` is 1 but `reg.shape` can be arbitrary, and we have one soquet for each `reg.shape` and one wire symbol for each soquet.